Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/lucene/impl/LuceneUtility.java

```diff
@@ -0,0 +1,302 @@
+            LuceneField luceneFieldAnnotation = field.getAnnotation(LuceneField.class);
+            field.setAccessible(true);
+            try {
+                if (luceneFieldAnnotation != null && d.get(luceneFieldAnnotation.lucenefieldName()) != null)
+                    field.set(con, d.get(luceneFieldAnnotation.lucenefieldName()));
+            } catch (IllegalArgumentException e) {
+                throw new IllegalArgumentException(e);
+            } catch (IllegalAccessException e) {
+                throw new IllegalAccessException();
+            }
+        }
+        con.setId(d.get("id"));
+        return con;
+    }
+
+    @Override
+    public void deleteIndexes() throws LuceneException {
+        try {
+            Query q = new QueryParser("conceptType", whiteSpaceAnalyzer).parse("*:*");
+            writer.deleteDocuments(q);
+            writer.commit();
+        } catch (IOException ex) {
+            throw new LuceneException("Issues in deleting wordnet concepts. Please retry", ex);
+        } catch (ParseException e) {
+            throw new LuceneException("Issues in framing the query", e);
+        }
+    }
+
+    protected void createDocuments(Iterator<IIndexWord> iterator, IDictionary dict, IndexWriter writer)
+            throws IOException {
+        for (; iterator.hasNext();) {
+            IIndexWord indexWord = iterator.next();
+            List<IWordID> wordIdds = indexWord.getWordIDs();
+
+            for (IWordID wordId : wordIdds) {
+
+                Document doc = new Document();
+                doc.add(new StringField(LuceneFieldNames.WORD, wordId.getLemma(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.POS, wordId.getPOS().toString(), Field.Store.YES));
+
+                IWord word = dict.getWord(wordId);
+                doc.add(new StringField(LuceneFieldNames.DESCRIPTION, word.getSynset().getGloss(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.ID, word.getID().toString(), Field.Store.YES));
+                doc.add(new StringField(LuceneFieldNames.WORDNETID, word.getID().toString(), Field.Store.YES));
+
+                ISynset synset = word.getSynset();
+                List<IWord> synonyms = synset.getWords();
+                StringBuffer sb = new StringBuffer();
+                for (IWord syn : synonyms) {
+                    if (!syn.getID().equals(word.getID()))
+                        sb.append(
+                                syn.getID().toString() + edu.asu.conceptpower.servlet.core.Constants.SYNONYM_SEPARATOR);
+                }
+                doc.add(new StringField(LuceneFieldNames.SYNONYMID, sb.toString(), Field.Store.YES));
+                // Adding this new data to delete only wordnet concepts while
+                // adding all wordnet concepts from jwi.
+                doc.add(new StringField(LuceneFieldNames.CONCEPTTYPE, "wordnetconcept", Field.Store.YES));
+                writer.addDocument(doc);
```

Comment:
I wouldn't throw the exception, handle it. otherwise if Lucene can't write one of the words, it'll give up on all of them. I'd return a result object that says how many words couldn't be indexed.